annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1291,238 @@ def test_get_page_number_combinations(
+            ],
+            ["b908ccf0-6eab-4b8e-afb7-913372ddf4e5"],
+            [
+                User(user_id="b908ccf0-6eab-4b8e-afb7-913372ddf4e5"),
+                User(user_id="00d37dc4-b7da-45e5-bcb5-c9b82965351c"),
+            ],
+        ),
+    ),
+)
+def test_prepare_users(
+    users_id: List[UUID], example_db: Tuple[UUID], expected_output: List[User]
+):
+    db = Mock()
+
+    def read_user_side_effect(db, user_id):
+        print("READ ME")
+        if user_id in example_db:
+            return User(user_id=user_id)
+        return None
+
+    def create_user_side_effect(db, user_id):
+        print("CREATE ME")
+        return User(user_id=user_id)
+
+    with patch(
+        "annotation.distribution.main.read_user",
+        side_effect=read_user_side_effect,
+    ), patch(
+        "annotation.distribution.main.create_user",
+        side_effect=create_user_side_effect,
+    ):
+        output = prepare_users(db, users_id)
+        for a, e in zip(output, expected_output):
+            assert a.user_id == e.user_id
+
+
+@pytest.mark.parametrize(
+    (
+        "users",
+        "all_job_pages_sum",
+        "average_job_pages",
+        "users_default_load",
+        "users_overall_load",
+        "expected_output",
+        "expected_share_loads",
+    ),
+    (
+        (
+            [{"user_id": "0", "overall_load": 10, "default_load": 10}],
+            10,  # all_job_pages_sum
+            10.0,  # average_job_pages
+            10,  # users_default_load
+            10,  # users_overall_load
+            1.0,  # expected_output
+            [1],  # expected_share_loads
+        ),
+    ),
+)
+def test_find_users_share_loads(
+    users: List[DistributionUser],
+    all_job_pages_sum: int,
+    average_job_pages: float,
+    users_default_load: int,
+    users_overall_load: int,
+    expected_output: float,
+    expected_share_loads: List[float],
+):
+    output = find_users_share_loads(
+        users,
+        all_job_pages_sum,
+        average_job_pages,
+        users_default_load,
+        users_overall_load,
+    )
+    assert output == expected_output
+    for user, expected_share_load in zip(users, expected_share_loads):
+        assert user["share_load"] == expected_share_load
+
+
+@pytest.mark.parametrize(
+    ("files", "user_pages", "expected_output"),
+    (
+        (
+            [{"pages_number": 1, "file_id": 0}],
+            1,
+            [{"pages_number": 1, "file_id": 0}],
+        ),
+        (
+            [
+                {"pages_number": 10, "file_id": 0},
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+            90,
+            [
+                {"pages_number": 20, "file_id": 1},
+                {"pages_number": 30, "file_id": 2},
+                {"pages_number": 40, "file_id": 3},
+            ],
+        ),
```

Comment:
you only change the actual number, let's simplify parametrize cases and construct full input/output data in the test

Reply:
I noticed that I needed to add another test case. Please tell me if you think I should still simplify the parametrize. The new test case makes it a bit more complicated

Reply:
I did not get the problem here. You can only provide numbers, it wont complicate the test

Reply:
Sorry again. Now I know what you meant by simplify the parametrize

Reply:
it is still the same
<img width="373" alt="image" src="https://github.com/user-attachments/assets/6bdf5e1b-e018-4404-9c65-188272124d5e">

Reply:
⁠Regarding this, I thought that i cant remove file_id's. If we have multiple files with the same page_number, then we should keep file_id's to check if the output was as expected. But if i want to just see if the tests work correctly, i could just:

`​assert sum([file["pages_number"] for file in output]) == user_pages`

In that case, file_id's are redundant and I'll remove them completely

Reply:
Nevermind that, I think I got it